<launch>
  <arg name="pose_topic" default="/camera_boom_pose" />
  <arg name="pose_control_service" default="/camera_boom_control" />

  <!-- Launch Motor Controller Driver -->
  <node pkg="hd_driver" type="hd_manager" name="aero_camera_boom_hd_manager">
    <param name="feedback_rate" value="5" />
    <param name="reference_frame" value="/boom_base" />
    <param name="port" value="/dev/BOOM" />
    <param name="control_service" value="hd_camera_boom_control" />
    <param name="info_topic" value="hd_camera_boom_feedback" />
    <param name="pause_topic" value="/pause" />
  </node>
  <!-- Launch Camera Boom Controller -->
  <node pkg="device_driver_base" type="motor_joint_position_controller" name="aero_camera_boom_node">
    <param name="ticks_per_radian" value="2086075.67009" />
    <param name="zero_tick_position" value="-110100" />
    <param name="src_frame" value="boom_base" />
    <param name="joint_name" value="boom_joint" />
    <param name="motor_control_service" value="hd_camera_boom_control" />
    <param name="joint_control_service" value="$(arg pose_control_service)" />
    <param name="motor_feedback_topic" value="hd_camera_boom_feedback" />
    <param name="is_angle_bounded" value="true" />
    <param name="min_angle" value="-3.2" />
    <param name="max_angle" value="3.2" />
    <param name="velocity_zero_threshold" value="0.001" />
  </node>
</launch>